setup.py

```diff
@@ -1,15 +1,48 @@
 from setuptools import setup, Extension
-import numpy as np
+
+class NumpyImport:
+  def __repr__(self):
+    import numpy as np
+
+    return np.get_include()
```

Comment:
I don't understand the point of this class. I think it's meant to defer the import of `numpy`, but you call `str(NumpyImport())` in the arguments to `setup`, so it will be evaluated and import `numpy` at startup anyway. Was this change made redundant by adding `pyproject.toml` with build system requires?

Reply:
Yes, the point of the script is to delay the import of numpy. Originally, importing numpy concurrently with installation would cause the setup script to crash as numpy would not be found. This was a solution I found on StackOverflow. 

Later on, new python versions would not properly render the __repr__, so it was necessary to add str to get it to evaluate properly, but you might be right that it has rendered the delay mechanism superfluous.

Reply:
@william-silversmith Thanks for explaining (and for contributing this solution).